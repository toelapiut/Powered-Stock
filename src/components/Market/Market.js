import React from 'react';
import MarketMargin from '../../components/MarketMargin';
import PropTypes from 'prop-types';
import styles from './styles.module.css';
export const Market = ({ticker,totalRevenue, companyName, fullCompanyName, openingMargin}) => {

  return (
    <div className={styles.container}  data-test={'market-container'}>
      <div className={styles.shares}>
        <h1>{ticker}</h1>
        <p className={styles.revenue}>{totalRevenue.toLocaleString()}</p>
      </div>
      <div className={styles.company}>
        <p className={styles.name}>{companyName.toLowerCase()}</p>
        <MarketMargin
          margin={openingMargin}
          fullCompanyName={fullCompanyName}
        />
      </div>
    </div>
  );
};

Market.propTypes = {
  ticker: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  fullCompanyName: PropTypes.string.isRequired,
  openingMargin: PropTypes.number.isRequired,
  totalRevenue: PropTypes.number.isRequired,
};