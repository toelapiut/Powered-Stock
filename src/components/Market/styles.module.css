.container {
}

.shares {
    @apply flex flex-row justify-between mb-2;
}

.ticker{
    @apply text-xl font-black
}

.revenue{
    @apply font-bold text-white;
}

.company{
    @apply flex flex-row justify-between items-center;
}

.name{
    @apply font-bold text-liberty text-sm capitalize;
}