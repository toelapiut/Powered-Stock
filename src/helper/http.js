import axios from 'axios';

export const quandl = axios.create({
  baseURL: process.env.process.env.REACT_APP_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    'Allow': 'GET, PUT, HEAD, OPTIONS'
  }
});

quandl.interceptors.request.use((config) => {
  config.params = config.params || {};
  config.params['key'] = process.env.REACT_APP_API_KEY;
  return config;
});
