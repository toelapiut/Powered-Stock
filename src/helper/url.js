export const url = {
  markets: 'datatables'
};

export const types = {
  ZACKS_CP: 'ZACKS/CP',
  ZACKS_MT: 'ZACKS/MT'
};