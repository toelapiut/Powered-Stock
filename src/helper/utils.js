import humps from 'humps';
import checkPropTypes from 'check-prop-types';
import {timeFormat} from 'd3-time-format';
import {bisector} from 'd3-array';
import moment from 'moment';

export const transpose = (rows, cols) => {
  /*
   * check if the row and col are empty and throw an error to
   * inform of the necessary requirements
   */
  if (typeof rows === 'undefined' || typeof cols === 'undefined') {
    throw new Error('row and col can not be undefined');
  }

  /*
   * throw error if the array isn't an array or col isn't an array
   */
  if (!Array.isArray(rows)) throw new Error('row must be  array');
  if (!Array.isArray(cols)) throw new Error('col must be  array');

  let dataset = [];

  for (let row of rows) {
    /*
     * check the data type before iterating through the matrix
     */
    if (!Array.isArray(row)) throw new Error('row must be a matrix row: [[a, b]]');
    let obj = {};
    row.forEach((item, index) => {
      obj[cols[index]['name']] = item;
    });
    dataset.push(obj);
  }

  return dataset;
};


export const transposeStock = (rows) => {
  let data = [];
  for (let row of rows) {
    let obj = {};
    const [date, close] = row;
    obj['date'] = date;
    obj['close'] = close;
    data.push(obj);
  }
  return data;
};

export const toCamelCase = (item) => {
  if (typeof item === 'undefined') {
    return item;
  }
  return humps.camelizeKeys(item);
};

export const checkPositivity = (num) => {
  if (typeof num === 'undefined') throw new Error('argument must not be empty');
  if (typeof num !== 'number') throw new Error('argument is not a number');
  let check = Math.sign(Number(num));
  return check !== -1;
};

export const formatDate = timeFormat('%b %d, \'%y');

export const getDate = (d) => new Date(d.date);

export const getStockValue = (d) => d.close;

export const bisectDate = bisector(d => new Date(d.date)).left;

export const getWindowDimensions = () => {
  const {innerWidth: width, innerHeight: height} = window;
  return {
    width,
    height
  };
};

export const dateFormat = (date, format = 'DD MMM, YYYY') => {
  return moment(date).format(format);
};

export const findTestByAttr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const checkProps = (component, expectedProps) => {
  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

