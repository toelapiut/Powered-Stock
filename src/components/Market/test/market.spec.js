import React from 'react';
import Market from '../';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {findTestByAttr, checkProps} from '../../../helper/utils';


const setUp = (props = {}) => {
  return shallow(<Market {...props}/>);
};

let props = {
  active: 'false',
  ticker: 'APPL',
  totalRevenue: 45244,
  companyName: 'Apple Inc',
  fullCompanyName: 'Apple Incorporation',
  openingMargin: -1.00
};

describe('Market Component', () => {

  describe('Component renders tests', () => {
    let component;
    beforeAll(() => {
      component = setUp(props);
    });

    test('should render snapshot correctly', () => {
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render without error', () => {
      const wrapper = findTestByAttr(component, 'market-container');
      expect(wrapper.length).toBe(1);
    });

  });

  describe('Check property types', () => {
    it('should not throw a warning', () => {
      const propErr = checkProps(Market, props);
      expect(propErr).toBeUndefined();
    });
  });

  describe('Have props', () => {

  });
});