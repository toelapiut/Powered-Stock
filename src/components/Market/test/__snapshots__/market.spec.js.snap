// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market Component Component renders tests should render snapshot correctly 1`] = `
<div
  className="container"
  data-test="market-container"
>
  <div
    className="shares"
  >
    <h1>
      APPL
    </h1>
    <p
      className="revenue"
    >
      45,244
    </p>
  </div>
  <div
    className="company"
  >
    <p
      className="name"
    >
      apple inc
    </p>
    <MarketMargin
      fullCompanyName="Apple Incorporation"
      margin={-1}
    />
  </div>
</div>
`;
